### PR TITLE
ROC-4098 Renaming payment option

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPaymentOption.java [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 public enum CCDPaymentOption {
     IMMEDIATELY("immediately"),
-    FULL_BY_SPECIFIED_DATE("on or before specified date"),
+    BY_SPECIFIED_DATE("on or before specified date"),
```
Comment (yarpo) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPaymentOption.java]: I suppose you need to update this as well:

src/integration-test/tests/citizen/ccj/pages/payment-options.ts:5

```
const fields = {
  paymentOption: {
    Immediate: 'input[id=optionIMMEDIATELY]',
    Instalments: 'input[id=optionINSTALMENTS]',
    FullBySetDate: 'input[id=optionFULL_BY_SPECIFIED_DATE]'
  }
}
```

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentOption.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 public enum PaymentOption {
     IMMEDIATELY("Immediately"),
+    BY_SPECIFIED_DATE("By a set date"),
     FULL_BY_SPECIFIED_DATE("By a set date"),
```
Comment (sabahirfan) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentOption.java]: I suppose this one we are keeping for backward compatibility only? 

Also on `citizen-frontend` we just have `FULL_BY_SPECIFIED_DATE` how that would work with new `BY_SPECIFIED_DATE` during JSON mapping in Request/Response?

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPaymentOption.java (hunk 1 above)
Comment (dharmendrak) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPaymentOption.java]: there is another PR for it mate

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentOption.java (hunk 2 above)
Comment (dharmendrak) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentOption.java]: get on paymentOption will always return BY_SET_DATE

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPaymentOption.java (hunk 1 above)
Comment (yarpo) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CCDPaymentOption.java]: why? It's the same git repo

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentOption.java (hunk 2 above)
Comment (sabahirfan) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentOption.java]: yes, that's what I meant and front-end will fail to map this new enum value

Comment (dharmendrak) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentOption.java]: there is a PR for frontend

Comment (sabahirfan) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/PaymentOption.java]: Great.

**Comment by sabahirfan** [comment]
Front-end PR: https://github.com/hmcts/cmc-citizen-frontend/pull/676

**Pull request merged**